metecho/api/serializers.py

```diff
@@ -361,26 +361,53 @@ def create(self, validated_data):
     def update(self, instance, validated_data):
         user = getattr(self.context.get("request"), "user", None)
         originating_user_id = str(user.id) if user else None
-        if instance.assigned_dev != validated_data["assigned_dev"]:
-            if validated_data.get("should_alert_dev"):
-                self.try_send_assignment_emails(instance, "dev", validated_data, user)
-            orgs = instance.scratchorg_set.active().filter(
-                org_type=SCRATCH_ORG_TYPES.Dev
-            )
-            for org in orgs:
-                org.queue_delete(originating_user_id=originating_user_id)
-        if instance.assigned_qa != validated_data["assigned_qa"]:
-            if validated_data.get("should_alert_qa"):
-                self.try_send_assignment_emails(instance, "qa", validated_data, user)
-            orgs = instance.scratchorg_set.active().filter(
-                org_type=SCRATCH_ORG_TYPES.QA
-            )
-            for org in orgs:
-                org.queue_delete(originating_user_id=originating_user_id)
+        self._handle_reassign(
+            "dev", instance, validated_data, user, originating_user_id
+        )
+        self._handle_reassign("qa", instance, validated_data, user, originating_user_id)
         validated_data.pop("should_alert_dev", None)
         validated_data.pop("should_alert_qa", None)
         return super().update(instance, validated_data)
 
+    def _handle_reassign(
+        self, type_, instance, validated_data, user, originating_user_id
+    ):
+        if (
+            getattr(instance, f"assigned_{type_}")
+            != validated_data[f"assigned_{type_}"]
+        ):
```

Comment:
@wlonk We're going through a lot of unnecessary logic here in the case where the assigned user has been removed. In that case I think we could bail early and just soft-delete the org.